### DISCUSSION TITLE
NextRouter was not mounted (Navigation Progress)

### DISCUSSION THREAD

**Discussion by whitigol** [vercel/next.js#46256] [Help]
I'm using MantineUI, and I'm trying to use the NavigationProgress component. But when trying to use `useRouter`, I'm getting the error `NextRouter was not mounted`.

Here's my code:
```typescript
import { useEffect } from 'react';
import { useRouter } from 'next/router';
import {
    NavigationProgress,
    startNavigationProgress,
    completeNavigationProgress,
} from '@mantine/nprogress';

export default function NavProgress() {
    const router = useRouter();

    useEffect(() => {
        const handleStart = (url: string) =>
            url !== router.asPath && startNavigationProgress();
        const handleComplete = () => completeNavigationProgress();

        router.events.on('routeChangeStart', handleStart);
        router.events.on('routeChangeComplete', handleComplete);
        router.events.on('routeChangeError', handleComplete);

        return () => {
            router.events.off('routeChangeStart', handleStart);
            router.events.off('routeChangeComplete', handleComplete);
            router.events.off('routeChangeError', handleComplete);
        };
    }, [router.asPath]);
    return <NavigationProgress />;
}
```

I'm using Next13 with the experimental App Directory. Is there something I'm doing incorrectly? Thanks in advance!

**Comment by icyJoseph** [comment]
Hi,

So this code is inside `app` right?

Inside `app` there's no `next/router`, but `next/navigation`:

- https://beta.nextjs.org/docs/api-reference/use-router

But that's not the bad news, the problem is that in `app` there's no router events anymore!

Gimme a few minutes to find some code that might help you out, as well as some reasoning as to why is there no route events anymore.

**Reply by icyJoseph** [reply]
Here ya go:

- https://github.com/vercel/next.js/discussions/42016#discussioncomment-3991279

**Reply by whitigol** [reply]
Thank you! Will take a look when home.
